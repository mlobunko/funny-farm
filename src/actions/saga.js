export const playCowSound = () => ({
  type: "PLAY_COW_SOUND"
});

export const playCatSound = () => ({
  type: "PLAY_CAT_SOUND"
});

export const playDogSound = () => ({
  type: "PLAY_DOG_SOUND"
});

export const playDuckSound = () => ({
  type: "PLAY_DUCK_SOUND"
});

export const playSheepSound = () => ({
  type: "PLAY_SHEEP_SOUND"
});

export const playChickenSound = () => ({
  type: "PLAY_CHICKEN_SOUND"
});

export const startPlayMusic = () => ({
  type: "START_PLAY_MUSIC"
});

export const changePlayMusic = () => ({
  type: "CHANGE_PLAY_MUSIC"
});
