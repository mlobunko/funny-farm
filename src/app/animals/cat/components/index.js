import Cat from './CatContainer';

export default Cat;
