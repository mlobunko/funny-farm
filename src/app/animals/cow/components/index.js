import Cow from './CowContainer';

export default Cow;
