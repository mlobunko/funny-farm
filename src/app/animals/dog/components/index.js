import Dog from './DogContainer';

export default Dog;
