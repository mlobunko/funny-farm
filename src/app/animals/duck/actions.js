import * as t from './actionTypes';

export const playSound = () => ({
  type: t.PLAY_SOUND
});

export const playingOn = () => ({
  type: t.PLAING_ON
});

export const playingOff = () => ({
  type: t.PLAYING_OFF
});
