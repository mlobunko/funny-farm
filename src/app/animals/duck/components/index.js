import Duck from './DuckContainer';

export default Duck;
