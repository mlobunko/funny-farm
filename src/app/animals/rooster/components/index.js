import Rooster from './RoosterContainer';

export default Rooster;
