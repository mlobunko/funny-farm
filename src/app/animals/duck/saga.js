import { delay } from 'redux-saga';
import { put, call, take, fork } from 'redux-saga/effects';

import clickedAnimal from 'shared/clickedAnimal';

import * as t from './actionTypes';
import * as actions from './actions';
import * as model from './model';
import { NAME } from './constants';

export function* watchPlaySound() {
  while (true) {
    yield take(t.PLAY_SOUND);
    yield put(actions.playingOn());
    yield put(clickedAnimal.actions.set(NAME));
    yield fork(model.playSound);
    yield call(delay, 700);
    yield fork(model.playSound);
    yield call(delay, 700);
    yield put(actions.playingOff());
  }
}
