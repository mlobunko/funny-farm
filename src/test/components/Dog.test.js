import React from "react";
import { shallow } from "enzyme";
import { Dog } from "../../components/Dog";
import { playDogSound } from "../../actions/saga";

it("should render Dog component correctly", () => {
  const wrapper = shallow(<Dog playDogSound={playDogSound} />);
  expect(wrapper).toMatchSnapshot();
});

it("should call playDogSound on buttonClick", () => {
  const playDogSound = jest.fn();
  const wrapper = shallow(<Dog playDogSound={playDogSound} />);
  wrapper.find(".dog__image").simulate("click");
  expect(playDogSound).toHaveBeenCalledTimes(1);
});
