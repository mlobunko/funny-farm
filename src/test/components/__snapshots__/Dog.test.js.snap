// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Dog component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    playDogSound={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="dog__image"
        onClick={[Function]}
/>,
      "className": "dog",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "dog__image",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="dog__image"
          onClick={[Function]}
/>,
        "className": "dog",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "dog__image",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
